client/src/app/pages/applications/analysis-wizard/utils.ts

```diff
@@ -73,34 +73,31 @@ export const toggleSelectedTargets = (
   return toggle(selectedTargets, target, (t) => t.id);
 };
 
-export const getUpdatedFormLabels = (
+export const updateSelectedTargetLabels = (
```

Comment:
the refactoring looks good but to be on the safe side I would create some unit tests before this PR as safety net.  This way we could verify that that after refactoring the flow is still the same.

Reply:
See #2325!